### DISCUSSION TITLE
Is there any example for Mohr-Coulomb yield criterion based perfect plasticity simulations

### DISCUSSION THREAD

**Discussion by Yulongggggg** [idaholab/moose#28925] [Q&A Modules: Solid mechanics] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

<recently I am using 'SolidMechanicsPlasticMohrCoulomb' and 'ComputeMultiPlasticityStress' to simulate the perfect plasticity simulation of tunneling process, but my numerical solution does not fit with the analytical solutions, so I was wonder if there's any example for this problem or I have misused the MOOSE objects? I have attached my code link below>

**Comment by Yulongggggg** [comment]


**Reply by GiudGiud** [reply]
You used this
        material_output_family = 'MONOMIAL'
        material_output_order = 'FIRST'
but the exodus output of first order monomial is not great (it's a difficult problem). Try constant monomials

could you try with this object instead of computeMultiplePlasticity?
https://mooseframework.inl.gov/source/materials/IsotropicPlasticityStressUpdate.html

**Reply by Yulongggggg** [reply]
Thank you! I was thinking if the constant monomials will decrease my computational accuracy. Because i have used constant monomials before and the precision is not promising. I see that the yield criterion used in [IsotropicPlasticityStressUpdate is von mises, and I am simulating the rock behavior so i'd better use the mohr-coulomb

**Reply by GiudGiud** [reply, marked as answer]
If it's just for output purposes then it does not matter what the precision is?

If you are using the material property output for the calculation we ll have to be a little careful

**Reply by Yulongggggg** [reply]
Thank you for helping, yeah, I might use it for further calculation. I have another question: since the mohr-coulomb yield criterion hasn't been documented yet, I was wondering what plastic potential function is used in the mohr-coulomb plasticity calculation, is it identical to the yield function or have different form?

**Reply by GiudGiud** [reply]
You are talking about the user objects for Mohr Coulomb?
@WilkAndy
